tools/rostest/src/rostest/rostest_parent.py

```diff
@@ -48,14 +49,37 @@
 
 class ROSTestLaunchParent(roslaunch.parent.ROSLaunchParent):
 
-    def __init__(self, config, roslaunch_files, port=0):
+    def __init__(self, config, roslaunch_files, port=0, local=False, clear=False):
         if config is None:
             raise Exception("config not initialized")
         # we generate a run_id for each test
-        run_id = roslaunch.core.generate_run_id()
+        if local:
+            param_server = rosgraph.Master('/roslaunch')
+            have_master = False
+            try:
+                run_id = param_server.getParam('/run_id')
+                have_master = True
+            except:
+                run_id = roslaunch.core.generate_run_id()
```

Comment:
The catch all here is a bit sketchy, maybe something like this would be more appropriate?:

```python
have_master = False
try:
    run_id = param_server.getParam('/run_id')
    have_master = True
except <Some expected exception> as e:
    rospy.logwarn(...)
<repeat as needed>
finally:
    if have_master is False:
        run_id = roslaunch.core.generate_run_id()
```

I'm worried the catch all might erroneously suppress some networking exception or some other unforeseen error.